src/server/hset_family.cc

```diff
@@ -166,7 +166,32 @@ OpStatus IncrementValue(optional<string_view> prev_val, IncrByParam* param) {
   param->emplace<int64_t>(new_val);
 
   return OpStatus::OK;
-};
+}
+
+void DoCleanup(StringMap* sm, DbSlice& db_slice, const DbContext& cntx, std::string_view key) {
+  if (!sm->Empty()) {
+    return;
+  }
+
+  if (auto it = db_slice.FindMutable(cntx, key, OBJ_HASH); it) {
+    it->post_updater.Run();
+    db_slice.Del(cntx, it->it);
```

Comment:
when we delete from database we need to write a journal change which will sent to replica
RecordJournal(op_args, "DEL"sv, ArgSlice{key});

Reply:
should this be added to places like https://github.com/dragonflydb/dragonfly/blob/f6ebfe0debe5e81ec1b53479c35a9c30e1214b43/src/server/hset_family.cc#L413 too? It is missing there.

Reply:
No. This OpDel is called from the transaction execution of hdel. The hdel command will be propagated to the replica part of the transaction framework. But in your flow the hgetall f.e if you delete the key you will need to manually call RecordJournal

Reply:
Got it, thanks. Will add journal change